src/chain/chaininterface.rs

```diff
@@ -54,44 +60,79 @@ pub enum ConfirmationTarget {
 /// called from inside the library in response to ChainListener events, P2P events, or timer
 /// events).
 pub trait FeeEstimator: Sync + Send {
-	fn get_est_sat_per_vbyte(&self, ConfirmationTarget) -> u64;
+	fn get_est_sat_per_vbyte(&self, confirmation_target: ConfirmationTarget) -> u64;
 }
 
 /// Utility to capture some common parts of ChainWatchInterface implementors.
 /// Keeping a local copy of this in a ChainWatchInterface implementor is likely useful.
 pub struct ChainWatchInterfaceUtil {
 	watched: Mutex<(Vec<Script>, Vec<(Sha256dHash, u32)>, bool)>, //TODO: Something clever to optimize this
 	listeners: Mutex<Vec<Weak<ChainListener>>>,
+	reentered: AtomicUsize
 }
 
-impl ChainWatchInterfaceUtil {
-	pub fn new() -> ChainWatchInterfaceUtil {
-		ChainWatchInterfaceUtil {
-			watched: Mutex::new((Vec::new(), Vec::new(), false)),
-			listeners: Mutex::new(Vec::new()),
-		}
-	}
-
-	pub fn install_watch_script(&self, spk: Script) {
+/// Register listener
+impl ChainWatchInterface for ChainWatchInterfaceUtil {
+	fn install_watch_script(&self, script_pub_key: Script) {
 		let mut watched = self.watched.lock().unwrap();
-		watched.0.push(Script::from(spk));
+		watched.0.push(Script::from(script_pub_key));
+		self.reentered.fetch_add(1, Ordering::Relaxed);
 	}
 
-	pub fn install_watch_outpoint(&self, outpoint: (Sha256dHash, u32)) {
+	fn install_watch_outpoint(&self, outpoint: (Sha256dHash, u32)) {
 		let mut watched = self.watched.lock().unwrap();
 		watched.1.push(outpoint);
+		self.reentered.fetch_add(1, Ordering::Relaxed);
 	}
 
-	pub fn watch_all_txn(&self) { //TODO: refcnt this?
```

Comment:
I believe this TODO was intended to point out that we may want to refcount the watch_all, and have something like an unwatch_all, but then you'd need to have a count of how many times it was called? Anyway, probably worth getting rid of...

Reply:
there is also no unregister. Let see if it is sufficient.